### DISCUSSION TITLE
Tool to automatically generate content stubs from specifications

### DISCUSSION THREAD

**Discussion by bershanskiy** [mdn/mdn-community#572] [Content] [answered]
I remember there was a discussion about generating the structure of content documents automatically from spec data. The idea was to have a Yoari command which would fetch IDL for a specific specification and would automatically generate "stub" documents for interfaces. The documents would basically have only headers and a few macros like `{{Specification}}`, `{{Compat}}`, and the like. Did that idea ever come to fruition?

**Comment by hamishwillee** [comment, marked as answer]
No. Though there are pre-created stubs in https://github.com/jpmedley/mdn-boilerplates . I doubt these are fully up to date to match the current page metadata etc. @jpmedley could confirm.

There is still much talk of this kind of thing though. Not so much for generating stubs, but for automation of adding things like "Secure context" and "available in workers" tagging.

**Reply by jpmedley** [reply]
Those are out of date. The necessary change to how events are processed broke my algorithm. I've been working on an update, but my duties at Google have not allowed me to spend as much time at it lately. I'm getting close though and hope to have an update soon.

By the way, I base those on Chrome IDL files rather than specs. Editors drafts are slightly ahead of implementation. Official specs are slightly behind because they may not yet reflect recent decisions by spec committees. It's possible to implement ahead of updating the spec. In other words, with specs, it's difficult to tell what should be built and what shouldn't. Basing on Chrome does't solve for situations where other browsers implement first, but it does solve for whether something is actually implemented. 

Regardless of what method is used, this is nowhere near as straightforward a problem as it seems. I'd be happy to meet with anyone who wants to work on this problem.
